src/app/core/components/loader/loader.component.ts

```diff
@@ -0,0 +1,48 @@
+import {
+    Component,
+    OnInit,
+    Input,
+    ChangeDetectionStrategy,
+    NgZone
+ } from '@angular/core';
+
+ import { LoaderService } from '../../services/loader.service';
+ import { Observable } from 'rxjs';
+
+@Component({
+    selector: 'app-loader',
+    providers: [],
+    styleUrls: [ './loader.component.scss' ],
+    templateUrl: './loader.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+
+export class LoaderComponent implements OnInit {
+    @Input() isPopupShow: Observable<any>;
```

Comment:
komponent ten podłączasz bez Input property w app.component:
```
  template: `
    <app-header></app-header>
```
Także, albo trzeba dodać inputs w templatcie, albo `isPopupShow` jest niepotrzebne

btw. możesz dodać input w templacie i podłączyć go z serwisem `loader.service` i na `ngOnChanges` pokazywać/chować loader - wtedy będzie Ci to działało z OnPush strategy.

popatrz jak ja to miałem:
https://github.com/przemko1984/angular2-epam/commit/e8c5869b969810448ec5f035acb661af75d60393
```
// glowny template
<loader-block [show]="loaderService.showLoader"></loader-block>

// loader template 
<div class="loader-container" [ngClass]="{show: show}">
  <div class="loader">Loading...</div>
</div> 

// loader component
ngOnChanges() {
 		this.loaderService.showLoader.subscribe(
             (resp) => {
                 this.show = resp;
             },
             (error) => {
                 console.error('error', error);
             }
         );
 }
```